### PR TITLE
feat(marketplace): add GLPI URL for checkpluginsupdate notif

### PR DESCRIPTION
**Pull request by flegastelois**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | N/A

### CODE REVIEW

File: install/empty_data.php [hunk 1]
```diff
@@ -4890,11 +4890,14 @@ public function getEmptyData(): array
 
 ##FOREACHplugins##
 ##plugin.name## :##plugin.old_version## -&gt; ##plugin.version##
-##ENDFOREACHplugins##',
+##ENDFOREACHplugins##
+
+##lang.glpi.url## :##glpi.url##',
```
Comment (cedric-anne) [inline install/empty_data.php:4895]:
```suggestion
##lang.glpi.url## : ##glpi.url##',
```

File: src/Marketplace/NotificationTargetController.php [hunk 2]
```diff
@@ -78,6 +78,9 @@ public function addDataForTemplate($event, $options = [])
             ];
         }
 
+        $this->data['##glpi.url##'] = $this->formatURL(NotificationTarget::GLPI_USER,
+                                                        '/front/marketplace.php');
```
Comment (cedric-anne) [inline src/Marketplace/NotificationTargetController.php:82]:
```suggestion
        $this->data['##glpi.url##'] = $this->formatURL(
            NotificationTarget::GLPI_USER,
            '/front/marketplace.php'
        );
```

File: src/Marketplace/NotificationTargetController.php [hunk 3]
```diff
@@ -122,7 +125,8 @@ public function getTags()
             'plugin.name'        => __('Plugin name'),
             'plugin.key'         => __('Plugin directory'),
             'plugin.version'     => __('Plugin new version number'),
-            'plugin.old_version' => __('Plugin old version number')
+            'plugin.old_version' => __('Plugin old version number'),
+            'glpi.url'           => __('URL of the application')
```
Comment (cedric-anne) [inline src/Marketplace/NotificationTargetController.php:129]:
```suggestion
            'glpi.url'           => __('URL of the application'),
```

**Review by trasher — APPROVED** [review]

**Review by cedric-anne — CHANGES_REQUESTED** [review]
There is already something that should be available globally: https://github.com/glpi-project/glpi/blob/d14f99ea0a807cb4be585a24cd39a2bbc7bb8952/src/NotificationTarget.php#L1333-L1336

I will try to investigate to see why it does not work.

**Comment by flegastelois** [comment]
> 



> There is already something that should be available globally:
> 
> https://github.com/glpi-project/glpi/blob/d14f99ea0a807cb4be585a24cd39a2bbc7bb8952/src/NotificationTarget.php#L1333-L1336
> 
> I will try to investigate to see why it does not work.

This is also what I had seen, but the "glpi.url" tag is not visible in the list of those available in the template.

File: src/Marketplace/NotificationTargetController.php [hunk 4]
```diff
@@ -122,7 +127,8 @@ public function getTags()
             'plugin.name'        => __('Plugin name'),
             'plugin.key'         => __('Plugin directory'),
             'plugin.version'     => __('Plugin new version number'),
-            'plugin.old_version' => __('Plugin old version number')
+            'plugin.old_version' => __('Plugin old version number'),
+            'glpi.url'           => __('URL of the application'),
```
Comment (cedric-anne) [inline src/Marketplace/NotificationTargetController.php:131]:
```suggestion
            'marketplace.url'    => __('URL of GLPI marketplace'),
```

File: src/Marketplace/NotificationTargetController.php [hunk 5]
```diff
@@ -78,6 +78,11 @@ public function addDataForTemplate($event, $options = [])
             ];
         }
 
+        $this->data['##glpi.url##'] = $this->formatURL(
```
Comment (cedric-anne) [inline src/Marketplace/NotificationTargetController.php:81]:
```suggestion
        $this->data['##marketplace.url##'] = $this->formatURL(
```

File: src/Marketplace/NotificationTargetController.php [hunk 6]
```diff
@@ -78,6 +78,11 @@ public function addDataForTemplate($event, $options = [])
...
+            NotificationTarget::GLPI_USER,
```
Comment (cedric-anne) [inline src/Marketplace/NotificationTargetController.php:82]: Not sure of this one, but all calls to `formatUrl` are using the same value
```suggestion
            $options['additionnaloption']['usertype'],
```

File: install/empty_data.php [hunk 7]
```diff
@@ -4890,11 +4890,14 @@ public function getEmptyData(): array
 
 ##FOREACHplugins##
 ##plugin.name## :##plugin.old_version## -&gt; ##plugin.version##
-##ENDFOREACHplugins##',
+##ENDFOREACHplugins##
+
+##lang.glpi.url## : ##glpi.url##',
```
Comment (cedric-anne) [inline install/empty_data.php:4895]:
```suggestion
##lang.marketplace.url## : ##marketplace.url##',
```

File: install/empty_data.php [hunk 8]
```diff
@@ -4890,11 +4890,14 @@ public function getEmptyData(): array
...
                 'content_html' => '&lt;p&gt;##lang.plugins_updates_available##&lt;/p&gt;
 &lt;ul&gt;##FOREACHplugins##
 &lt;li&gt;##plugin.name## :##plugin.old_version## -&gt; ##plugin.version##&lt;/li&gt;
-##ENDFOREACHplugins##&lt;/ul&gt;'
+##ENDFOREACHplugins##&lt;/ul&gt;
+&lt;p&gt;##lang.glpi.url## : &lt;a title="##lang.glpi.url##" href="##glpi.url##" target="_blank" rel="noopener"&gt;##glpi.url##&lt;/a&gt;&lt;/p&gt;'
```
Comment (cedric-anne) [inline install/empty_data.php:4900]:
```suggestion
&lt;p&gt;##lang.marketplace.url## : &lt;a title="##lang.marketplace.url##" href="##marketplace.url##" target="_blank" rel="noopener"&gt;##marketplace.url##&lt;/a&gt;&lt;/p&gt;'
```

**Review by cedric-anne — CHANGES_REQUESTED** [review]

**Comment by flegastelois** [comment]
Linked to https://github.com/glpi-project/glpi/pull/13830

**Review by orthagh — APPROVED** [review]

**Review by trasher — APPROVED** [review]

**Pull request merged**